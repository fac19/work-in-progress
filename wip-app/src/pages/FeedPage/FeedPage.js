import React, { useEffect } from "react";
import { feedPage } from "../../utils/get-fetch";
import ProjectFeed from "../../components/ProjectFeed/ProjectFeed";
import { PageHeading } from "../page.style";

const FeedPage = (props) => {
  const [projects, setProjects] = React.useState(null);
  const [isEmpty, setIsEmpty] = React.useState(false);

  useEffect(() => {
    feedPage().then((projects) => {
      setProjects(projects);
      if (projects.length === 0) {
        setIsEmpty(true);
      }
    });
  }, []);

  const isLoading = !projects;

  if (isLoading) {
    return <PageHeading>Loading your feed...</PageHeading>;
  } else if (isEmpty) {
    return (
      <PageHeading>
        You don't follow any projects yet...check them out in Explore!
      </PageHeading>
    );
  } else {
    return <ProjectFeed projects={projects}></ProjectFeed>;
  }
};

export default FeedPage;
