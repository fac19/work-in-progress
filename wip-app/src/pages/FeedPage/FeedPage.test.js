import React from "react";
import { createMemoryHistory } from "history";
import TestRenderer from "react-test-renderer";
import FeedPage from "./FeedPage";
import { Router } from "react-router-dom";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const history = createMemoryHistory();
    const component = TestRenderer.create(
      <Router history={history}>
        <FeedPage />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
