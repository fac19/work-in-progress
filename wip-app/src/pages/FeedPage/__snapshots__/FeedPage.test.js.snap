// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<h1
  className="sc-AxheI NtAGp"
>
  Loading your feed...
</h1>
`;
